x-pack/plugins/security_solution/public/management/components/endpoint_execute_action/execute_action_host_response_output.tsx

```diff
@@ -99,42 +96,72 @@ interface ExecuteActionHostResponseOutputProps {
 }
 
 export const ExecuteActionHostResponseOutput = memo<ExecuteActionHostResponseOutputProps>(
-  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize }) => {
+  ({ action, agentId = action.agents[0], 'data-test-subj': dataTestSubj, textSize = 'xs' }) => {
+    const { canWriteExecuteOperations } = useUserPrivileges().endpointPrivileges;
+
+    const [executeOutputContent, setExecuteOutputContent] = useState<
+      undefined | ResponseActionExecuteOutputContent
+    >(undefined);
+
     const outputContent = useMemo(
       () =>
         action.outputs &&
         action.outputs[agentId] &&
         (action.outputs[agentId].content as ResponseActionExecuteOutputContent),
-      [agentId, action]
+      [action.outputs, agentId]
     );
 
+    const { data: actionDetails, isFetching } = useGetActionDetails(action.id, {
```

Comment:
is this component being used from the Responder console?  if so, then do we have duplicate api calls being made to Action Details?

Wondering if this component should instead take in the `ActionDetails` as a input prop rather than to call teh API to get it.

Reply:
I believe there should be no duplicate API call: it should be handled (cached) by `react-query`, using the `action.id` as `queryKey`

Reply:
Quite right @gergoabraham. The component does actually take in `ActionDetails` as a prop, but since the data is coming in from the action list there is no `outputs`. So here (for execute output only) we are fetching the action details using the action details API that does have `outputs`. 

Another approach would be to make a fetch request when we expand any item in the list.